test/pytest/service-desktop/test_threads.py

```diff
@@ -1,45 +1,130 @@
+        except TransactionError:
+            return False
+        else:
+            return True, message
+
+    def delete_message_by_id(self, message_record_id):
+        try:
+            DeleteMessageById(message_record_id).run(self.harness)
+        except TransactionError:
+            return False
+        else:
+            return True
+
+
+@pytest.mark.service_desktop_test
+@pytest.mark.usefixtures("phone_unlocked")
+def test_marking_thread_as_read(harness):
+    message_number_1 = "123456789"
+    message_1_body = "Test message nr 1"
+
+    threads_tester = ThreadsTester(harness)
+
+    result, message_1_record = threads_tester.add_message(message_number_1, message_1_body)
+    assert result, "Failed to add message!"
+    result, received_message_1_record = threads_tester.get_message_by_id(message_1_record["messageID"])
+    assert result, "Failed to get message by id!"
+
+    assert threads_tester.mark_thread_as_unread(received_message_1_record["threadID"], False)
+    assert threads_tester.mark_thread_as_unread(received_message_1_record["threadID"], True)
+
+    assert threads_tester.delete_message_by_id(message_1_record["messageID"]), "Failed to delete a message!"
 
 
 @pytest.mark.service_desktop_test
 @pytest.mark.usefixtures("phone_unlocked")
-def test_threads(harness):
-    # getting all threads
-    body = {"category": "thread"}
-    ret = harness.endpoint_request("messages", "get", body)
-    assert ret["status"] == status["OK"]
-
-    # getting a number of threads
-    number_of_requested_threads = 3
-    if ret["body"]["totalCount"] < number_of_requested_threads:
-        number_of_requested_threads = ret["body"]["totalCount"]
-    body = {"category": "thread", "limit": number_of_requested_threads, "offset": 0}
-    ret = harness.endpoint_request("messages", "get", body)
-    assert ret["status"] == status["OK"]
-
-    threads = ret["body"]["entries"]  # getting entries
-    threads_count = len(threads)
-    assert threads_count == number_of_requested_threads
-
-    # Check if all fields are present and have proper type
-    message_types = [1, 2, 4, 8, 16, 18, 255]
-    print(ret["body"])
-    for thread in ret["body"]["entries"]:
-        print(thread)
-        assert type(thread["contactID"]) == int
+def test_getting_threads(harness):
+    message_number_1 = "123456789"
+    message_number_2 = "223456789"
+    message_number_3 = "323456789"
+    message_number_4 = "423456789"
+    message_number_5 = "523456789"
+    message_number_6 = "623456789"
+    message_1_body = "Test message nr 1"
+    message_2_body = "Test message nr 2"
+    message_3_body = "Test message nr 3"
+    message_4_body = "Test message nr 4"
+    message_5_body = "Test message nr 5"
+    message_6_body = "Test message nr 6"
+
+    threads_tester = ThreadsTester(harness)
+
+    result, message_1_record = threads_tester.add_message(message_number_1, message_1_body)
```

Comment:
why copy instead of for loop?

Reply:
as @pholat mentioned, this could be wrapped in a loop

Reply:
Loop added